contracts/PositionManager.sol

```diff
@@ -249,6 +253,24 @@ contract PositionManager is IPositionManager, ERC721Holder {
         _;
     }
 
+    ///@notice modifier to check if the position is owned by the positionManager
+    modifier onlyOwnedPosition(uint256 tokenId) {
+        StorageStruct storage Storage = PositionManagerStorage.getStorage();
+        try
```

Comment:
a simply require  INonfungiblePositionManager(Storage.uniswapAddressHolder.nonfungiblePositionManagerAddress()).ownerOf(
                tokenId
            ) == address(this) should work the same